source/authentication/apple.txt

```diff
@@ -59,379 +59,607 @@ following:
-         Press the :guilabel:`Next` button.
+               .. important:: Save the Key
 
-      #. Click :guilabel:`Continue` and then click :guilabel:`Save`.
-         Confirm that you have correctly configured the Services ID and
-         then click :guilabel:`Register`.
+                  You can only download the key one time. Make sure that you save
+                  the key somewhere safe in case you need it again. If you lose
+                  the key, you will need to generate a new one.
 
 
-   .. step:: Create a Private Key
+         .. step:: Create the Client Secret JWT
 
-      The client secret for Sign in with Apple is a JSON Web Token that you
-      create and sign with a **private key**. You need to generate the
-      private key through the Apple Developer Portal.
+            You can now create the client secret JWT for the Apple ID
+            authentication provider. Make sure that you have the following
+            information:
 
-      1. Click :guilabel:`Keys` in the left navigation menu.
+            - The :guilabel:`Bundle ID` of your app.
 
-      #. Click the blue plus icon next to :guilabel:`Keys`.
+            - The :guilabel:`Key ID` of the key that you created and the ``.p8``
+               file that contains the key.
 
-      #. On the :guilabel:`Register a New Key` page, enter a descriptive
-         :guilabel:`Key Name` and then scroll down to find the
-         :guilabel:`Sign in with Apple` row. Check the checkbox to enable
-         Sign in with Apple and then click :guilabel:`Configure`.
+            - Your Apple Team ID. You can find this in the top right of the Apple
+               Developer Portal.
 
-      #. On the :guilabel:`Configure Key` page, select the App ID that you
-         created as the :guilabel:`Primary App ID` and then click
-         :guilabel:`Save`.
+            .. figure:: /images/apple-team-id.png
+               :alt: An Apple team ID in the Apple Developer Portal
 
-         .. figure:: /images/apple-auth-configure-key.png
-            :alt: The Configure Key page in the Apple Developer Portal
+            Once you've confirmed that you have all the required information, you can use
+            a script to generate the JWT. You may define your own script or use the
+            script in this step.
 
-      #. Click :guilabel:`Continue` to review your key configuration. When
-         you're sure that you've configured the key correctly, click
-         :guilabel:`Register`.
+            To generate the JWT, we'll use the :rubygems:`jwt <gems/jwt>` gem. To install
+            it, run the following:
 
-      #. Copy the :guilabel:`Key ID` value somewhere that you can access it
-         later and then click :guilabel:`Download` to download the key as a
-         ``.p8`` text file. You will use these to generate the client
-         secret.
+            .. code-block:: shell
+
+               gem install jwt
```

Comment:
Does it matter which directory I run this from?

Reply:
Nope!